extensions/azcli/src/az.ts

```diff
@@ -41,19 +43,30 @@ export interface IAzTool extends azExt.IAzApi {
  */
 export class AzTool implements azExt.IAzApi {
 
-	private _semVersion: SemVer;
+	private _semVersionAz: SemVer;
+	private _semVersionArc: SemVer;
 
-	constructor(private _path: string, version: string) {
-		this._semVersion = new SemVer(version);
+	constructor(private _path: string, versionAz: string, versionArc: string) {
+		this._semVersionAz = new SemVer(versionAz);
+		this._semVersionArc = new SemVer(versionArc);
 	}
 
 	/**
-	 * The semVersion corresponding to this installation of az. version() method should have been run
+	 * The semVersion corresponding to this installation of Azure CLI. version() method should have been run
 	 * before fetching this value to ensure that correct value is returned. This is almost always correct unless
 	 * Az has gotten reinstalled in the background after this IAzApi object was constructed.
 	 */
-	public async getSemVersion(): Promise<SemVer> {
-		return this._semVersion;
+	public async getSemVersionAz(): Promise<SemVer> {
```

Comment:
I think they have to be async because before this is called we are finding Az and Arc and validating it. See api.ts:
getSemVersionAz: async () => {
			await localAzDiscovered;
			throwIfNoAz(azToolService.localAz);
			return azToolService.localAz.getSemVersionAz();

Reply:
Does that make sense? Let me know if not and I'll unmark as resolved, but resolving for now

Reply:
Ok, yeah if you're waiting on that promise then this is fine.